src/query/functions/temporal/aggregation.go

```diff
@@ -192,3 +220,62 @@ func sumAndCount(values []float64) (float64, float64) {
 
 	return sum, count
 }
+
+func makeQuantileOverTimeFn(scalar float64) aggFunc {
+	return func(values []float64) float64 {
+		valuesSlice := make(valsSlice, 0, len(values))
+		for _, v := range values {
+			valuesSlice = append(values, v)
+		}
+
+		return quantile(scalar, valuesSlice)
+	}
+}
+
+type valsSlice []float64
+
+func (s valsSlice) Len() int {
+	return len(s)
+}
+
+func (s valsSlice) Less(i, j int) bool {
+	if math.IsNaN(s[i]) {
+		return true
+	}
+
+	return s[i] < s[j]
+}
+
+func (s valsSlice) Swap(i, j int) {
+	s[i], s[j] = s[j], s[i]
+}
+
+// qauntile calculates the given quantile of a slice of values.
+//
+// This slice will be sorted.
+// If 'values' has zero elements, NaN is returned.
+// If q<0, -Inf is returned.
+// If q>1, +Inf is returned.
+func quantile(q float64, values valsSlice) float64 {
+	if len(values) == 0 {
+		return math.NaN()
+	}
+	if q < 0 {
+		return math.Inf(-1)
+	}
+	if q > 1 {
+		return math.Inf(+1)
+	}
+	sort.Sort(values)
```

Comment:
There can be a problem with NaNs actually; we don't want them to pollute the output since they don't exist in Prom's implementation. Instead of just doing a `sort` here, it would be better to put the through a sanitize function that will remove NaNs from the values, then sort.

On the bright side, this means we can use the inbuilt int slice sorter, and not have to worry about the `valsSlice` type